# Method 1

In [21]:
num=int(input("Enter a number which you want to check: "))
count=0

if num>1:
    for i in range(1,num+1):
        if (num%i)==0:
            count=count+1
    if count == 2:
            print("Prime number")
            
    else:
            
  
            print("Not prime number")
           

Enter a number which you want to check: 4
Not prime number


In [30]:
num=int(input("Enter a number which you want to check: "))
count=0

for i in range(1,num+1):
    if (num%i) == 0:
        count=count+1
if count == 2:
    print("Prime Number")

else:
    print("Not Prime number")



Enter a number which you want to check: 193
Prime Number


# Method 2

In [33]:
num=17

for i in range(2,num):
    if num%i == 0:
        print("Not Prime")
        break
else:
    print("Prime")

Prime


The first few prime numbers are as follows: 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97, 101, 103, 107, 109, 113, 127, 131, 137, 139, 149, 151, 157, 163, 167, 173, 179, 181, 191, 193, 197, and 199, etc.

In [39]:
num=int(input("Enter a number which you want to check: "))

for i in range(2,num):
    if num%i == 0:
        print("Not Prime")
        break
else:
    print("Prime")

Enter a number which you want to check: 13
Prime


In [41]:
num=int(input("Enter a number which you want to check: "))

for i in range(2,num):
    if num%i == 0:
        print("Not Prime")
        break
else:
    print("Prime")

Enter a number which you want to check: 59
Prime


# Method 3

In [44]:
num=151
prime=True

for i in range(2,num):
    if num % i == 0:
        prime= False
    
if prime:
    print("This is Prime Number.")
else:
    print("This is not a Prime Number")

This is Prime Number.


In [46]:
number=int(input("Enter a number: "))
prime=True

for i in range(2,number):
    if number % i == 0:
        prime=False
        
        
if prime:
    print(number,"is a prime number.")
else:
    print(number,"is not a prime number.")

Enter a number: 45
45 is not a prime number.


In [47]:
number=int(input("Enter a number: "))
prime=True

for i in range(2,number):
    if number % i == 0:
        prime=False
        
        
if prime:
    print(number,"is a prime number.")
else:
    print(number,"is not a prime number.")

Enter a number: 199
199 is a prime number.


# Write a program to print prime numbers from 1 to 20.

In [56]:
lower_range=int(input("Enter a lower range: "))
upper_range=int(input("Enter a upper range: "))

for num in range(lower_range,upper_range+1):  # for num in range(1,20);  num=1 to 20
    if num>1:                                # 1 is !> 1 therefore it take that is num=2 which is > 1.
        for i in range(2,num):                 # for 1st iteration num=2
            if num%i == 0:                
                break
        else:
            print(num)
   
                

  

Enter a lower range: 1
Enter a upper range: 20
2
3
5
7
11
13
17
19


In [57]:
2%2

0

# Write a program to print prime numbers from 5 to 20.

In [58]:
lower_range=int(input("Enter a lower range: "))   # 5
upper_range=int(input("Enter a upper range: "))   # 20

for num in range(lower_range,upper_range+1):  # for num in range(5,20);  num=5 to 20
    if num>1:                                # In num for 1st iteration 5 so 5>1;True.
        for i in range(2,num):                 # for 1st iteration num=5 therefore; i=2,3 & 4.
            if num%i == 0:                #  for 1st iteration 5 % 2 is !=0,5% 3is !=0 then 5% 4 is also != 0
                                              # for 2nd iteration num=6 and i = 2 to 5 ;6%2 = 0 therefore break the statement.
                                                # for 3rd itera. num = 7 and i=2 to 6 like wise.
                break        # This is for breaking the statement.
        else:
            print(num)   # print(num)
   
                


Enter a lower range: 5
Enter a upper range: 20
5
7
11
13
17
19


In [59]:
6%2

0

In [60]:
5%2

1

In [61]:
8%2

0

In [62]:
8//2

4

# Write a program to print prime numbers between 20 to 80.


In [63]:
lower=int(input("Enter lower range: "))
upper=int(input("Enter upper range: "))

for i in range(lower,upper):
    if i>1:
        for j in range(2,i):
            if (i%j)==0:
                break
        else:
            print(i)

Enter lower range: 20
Enter upper range: 80
23
29
31
37
41
43
47
53
59
61
67
71
73
79


# Write a program to print all prime numbers between 1 to 100.

In [6]:
for num in range(1,101):
    if num>1:
        for i in range(2,num):
            if (num%i)==10:
                break
                
else:
    print(num)
        

100


In [8]:
lower_range=1
upper_range=100

for num in range(lower_range,upper_range+1):
    if num > 1:
        for i in range(2,num):
            if num%i == 0:
                break
        else:
            print(num,end="\t")

2	3	5	7	11	13	17	19	23	29	31	37	41	43	47	53	59	61	67	71	73	79	83	89	97	

# Write a program to print all prime numbers between 100 to 200.

In [10]:
num1=100
num2=200

for i in range(num1,num2+1):
    if i > 1:
        for j in range(2,i):
            if (i%j)==0:
                break
                
        else:
            print(i)

101
103
107
109
113
127
131
137
139
149
151
157
163
167
173
179
181
191
193
197
199


# Write a program to check whether the given number is prime or not.


In [22]:
num=37
is_prime=True

for i in range(2,num):
    if num%i == 0:
        is_prime=False
        
if is_prime:
    print("Prime")
else:
    print("Not Prime")
        

Prime


# user input

In [27]:
num=int(input("Enter a number:"))
is_prime=True

for i in range(2,num):
    if (num%i) == 0:
        is_prime=False
        
if is_prime:
    print(num,"is prime number.")
else:
    print(num,"is not prime number.")

Enter a number:167
167 is prime number.


In [28]:
num=int(input("Enter a number:"))
is_prime=True

for i in range(2,num):
    if (num%i) == 0:
        is_prime=False
        
if is_prime:
    print(num,"is prime number.")
else:
    print(num,"is not prime number.")

Enter a number:193
193 is prime number.


In [35]:
num=int(input("Enter a number:"))
is_prime=True

for i in range(2,num):
    if (num%i) == 0:
        is_prime=False
        
if is_prime:
    print(num,"is prime number.")
else:
    print(num,"is not prime number.")

Enter a number:85
85 is not prime number.


In [43]:
num=int(input("Enter number:"))
count=0

for i in range(1,num+1):
    if (num%i) == 0:
        count=count+1
if count==2:
    print("Prime number.")
else:
    print("Not prime number.")

Enter number:149
Prime number.


In [42]:
num=int(input("Enter number:"))
count=0

for i in range(1,num+1):
    if (num%i) == 0:
        count=count+1
if count==2:
    print("Prime number.")
else:
    print("Not prime number.")

Enter number:157
Prime number.


In [44]:
num=int(input("Enter number:"))
count=0

for i in range(1,num+1):
    if (num%i) == 0:
        count=count+1
if count==2:
    print("Prime number.")
else:
    print("Not prime number.")

Enter number:84
Not prime number.
